library/sha256.c

```diff
@@ -389,10 +397,16 @@ static const unsigned char sha256_test_sum[6][32] =
 int mbedtls_sha256_self_test( int verbose )
 {
     int i, j, k, buflen, ret = 0;
-    unsigned char buf[1024];
+    unsigned char *buf;
```

Comment:
I agree with Janos, we cater to a large number of platforms and I am not sure we want to lose the ability to simple allocate these buffers in the stack. However, if we want to go that way, a clean way to do it remains to be decided...